crates/tabby-scheduler/src/lib.rs

```diff
@@ -8,14 +8,13 @@ mod indexer;
 use async_stream::stream;
 pub use code::CodeIndexer;
 use crawl::crawl_pipeline;
-use doc::SourceDocument;
+pub use doc::{create_web_index, SourceDocument};
```

Comment:
I recommend extract apis like `DocIndexer` to avoid exposing `SourceDocument` / `Indexer` to public. You might refer `CodeIndexer` as example

Reply:
Please create new type for DocIndexer instead of exposing SourceDocument directly